proto/config/edge_node_cluster.proto

```diff
@@ -0,0 +1,37 @@
+// Copyright(c) 2024 Zededa, Inc.
+// All rights reserved.
+
+syntax = "proto3";
+
+package org.lfedge.eve.config;
+option go_package = "github.com/lf-edge/eve-api/go/config";
+option java_package = "org.lfedge.eve.config";
+
+message EdgeNodeCluster {
+  // cluster name, in case it has multiple cluster on the same site
+  string ClusterName = 1;
+
+  // cluster UUID
+  string ClusterID = 2;
+
+  // Cluster-Interface
+  string ClusterInterface = 3;
+
+  // the 'cluster-prefix' IP address of the 'Cluster-Interface', 10.244.244.2/28
+  string ClusterIPPrefix = 4;
+
+  // This device is 'seed-server'
+  bool IsSeedServer = 5;
```

Comment:
I thought we wouldn't need this. Or will the controller pick a seed server without the user being involved?

Reply:
Yes, we decided to let controller to automatically pick, w/o user to choose. But the entry is needed.

Reply:
_This comment disappeared since the field was renamed `is_seed_server` in snake case (which is correct)._

The field `is_seed_server` is a one-time thing, yet it is a permanent part of the message `EdgeNodeCluster`. That makes it a short-lived-usefulness-and-meaning yet perpetually-lingering field. it could get confusing over time.
It also doesn't align as much with the language used for k3s.

Would it help to change the field to `cluster_init`? That aligns with how k3s uses it; "node, initialize a cluster instead of joining a cluster". And perhaps a separate `role` field, with `control-plane` and `worker`? We don't use `worker` yet, but we will.

k3s also ignores `--cluster-init` if it already has a local cluster.

That doesn't solve the lingering field, but it helps a bit.

Similar (but slightly different) question about [this field](https://github.com/lf-edge/eve-api/pull/46/files#diff-75211aa7f801986e27d4a3bdea12bd1c296af83e79750487bf1c7cb20a8c7e1aR32-R33):

```proto
  // SSIP = Seed-ServerIP address to join during bootstrap. E.g. 10.244.244.1
  string seed_server_ip = 7;
```

Is this just during _cluster_ bootstrap? In the future, we will have new nodes join a cluster, they will need to be told where to connect at _node_ bootstrap time. The k3s flag is called `--server`, kubeadm uses the field `apiServerEndpoint`, kubelet uses kubectl's `server` field.

Maybe it should be called `joinServer` or similar?

1. I start a single-node cluster (seed server)
2. I have 2 other nodes join - they don't know anything about seed server, just an IP to join
3. a month goes by
4. I ask 2 more nodes to join the cluster - the IP might be the original seed server, it might be something else entirely; seed has no meaning at this point anyways.

Reply:
Ok, I think for the EVE API, this isSeedServer is sort of redundant, in the way that
we have the 'cluster_ip_prefix', and the seed_server_ip items. If the seed_server_ip
matches the IP address of the 'cluster_ip_prefix', then the device can perform the
initial bootstrap functions. Otherwise the device will just do the join the the seed_server_ip address.
In terms of the name, 'join_server_ip' or 'bootstrap_server_ip', or we can think of a better one.

Yes, this node might be down and unavailable later for 4th or 5th node to join, but someone needs to
let the new node to know which IP address to use to join. So, in that sense, this 'join_server_ip' can
possibly change later when there is a new node to join.

The instruction has to come from the controller since the new node can not use the kubernetes resource
before it joins kubernetes.
We/controller either needs to do this 'automatically' or user has to do it manually, but we just decided we don't
want user to do manually at least for the initial cluster bring up.

Reply:
is_seed_server has been deleted, and seed_server_ip has been changed to join_server_ip.

Reply:
That makes sense, @famleebob . And based on @naiming-zededa's description, this gets sent (or looked at) when a node first joins (or initializes) a cluster? So the possible scenarios that an edge device can have are:

* I am joining an existing cluster as a control plane node - I see `join_ip` and `is_agent=false`, and use that IP to join the cluster using `cluster_token` 
* I am joining an existing cluster as a worker node - I see `join_ip` and `is_agent=true`, and use that IP to join the cluster using `cluster_token` 
* I am initializing a new cluster - I see a `join_ip` and `is_agent=false` and `cluster_ip_prefix=join_ip` so I know to initialize
* I already am part of a cluster - ignore

That makes sense.